### PR TITLE
Signup: user testing flow

### LINKED ISSUE #37659: Add prominent “Skip” buttons on plans and domains in sign-up.
If a customer isn’t ready to commit, we should still let them through to see if we can show value as they build their site.

This will give us some data and satisfy the spirit of “Starting for free”. (See Free option above. We will need to exclude a “free plan” option when a domain is selected.

This change should be included in user tests, but should be A/B tested before being launched.

Domains screen:
Screen Shot 2019-11-17 at 9 53 37 PM

Plans Screen:
Screen Shot 2019-11-17 at 9 53 49 PM

### PR DESCRIPTION
**Pull request by michaeldcain**
Signup parts of: paObgF-GK-p2

This edits the `test-fse` Signup flow for our next round of user testing. Edits include:

- Remove site type and site topic steps
- Set pub/maywood as the theme default when one isn't passed via query string or step dependency
- Move the domains skip button to the top of the domains step (matching the plans step)

There was some wonkiness on using a passwordless user step that I believe needs to wait for #37697.

Screen Shot 2019-11-13 at 10 10 17 AM

**To test:**
- visit `/start/test-fse/`
- steps should be: user, site title, domains (with top skip), plans
- verify that completing the flow results in a new site with the Maywood theme
- verify that skipping Signup auto-assigns a subdomain
- verify that a site title input in Signup is reflected on the end site
- visit `/start/onboarding/`
- complete Signup as normal (by picking a site type and topic, domain, and plan)
- verify that the resulting site looks as expected

FIxes #37659

### CODE REVIEW

File: client/lib/signup/site-type.js [hunk 1]
```diff
@@ -87,7 +87,7 @@ export function getAllSiteTypes() {
 			theme: 'pub/maywood',
 			designType: 'blog',
 			siteTitleLabel: i18n.translate( "Tell us your blog's name" ),
-			siteTitlePlaceholder: i18n.translate( "E.g., Stevie's blog " ),
+			siteTitlePlaceholder: i18n.translate( "E.g., Stevie's blog" ),
```
Comment (a8ci18n) [inline client/lib/signup/site-type.js]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

**Comment by andrewserong** [comment]
This is testing well for me @michaeldcain — the only issue is the default starter content is Restaurant specific because it's using the theme annotation. I wonder if there's a more neutral vertical we could pre-select for this one?

With the passwordless signup, what was the wonkiness you ran into? I've tried mocking it up locally, and I could get it working by replacing the [if statement in SignupForm](https://github.com/automattic/wp-calypso/blob/c6f5c2a6334ba0aeba8de104330896eb20e4f58a/client/blocks/signup-form/index.jsx#L958) with:

```js
                if (
                       this.props.flowName === 'test-fse' ||
                       ( this.props.flowName === 'onboarding' && 'passwordless' === abtest( 'passwordlessSignup' ) )
                ) {
```

and replacing the [conditional in the flow controller _processStep](https://github.com/automattic/wp-calypso/blob/6a3a080cc3760998fe909ea97827aad9483f887c/client/lib/signup/flow-controller.ts#L309) with:

```js
		if (
			( 'test-fse' === this._flowName && get( step, 'isPasswordlessSignupForm' ) ) ||
			( 'onboarding' === this._flowName &&
				get( step, 'isPasswordlessSignupForm' ) &&
				'passwordless' === abtest( 'passwordlessSignup' ) )
		) {
```

(I didn't want to commit this to the PR in case I've missed something!)

**Comment by p-jackson** [comment]
> There was some wonkiness on using a passwordless user step that I believe needs to wait for #37697.
No need to wait for this @michaeldcain, the only issue is the user in the test wouldn't be assigned a recaptcha score.

**Comment by michaeldcain** [comment]
@andrewserong @p-jackson thanks for the reviews - the "wonkiness" with passwordless and reCaptcha was an error about the reCaptcha key not being available in `config/_shared.json`. I'll try it again, but I wanted to get this reviewed for the next user test before then.

> I wonder if there's a more neutral vertical we could pre-select for this one?

I was wondering the same. I wonder if the best flow with @jblz's checklist change is to create a blank site without any vertical content.

**Comment by p-jackson** [comment]
That's strange. The key is defined in the development and horizon environments, so it shouldn't have had an issue getting it.

**Comment by apeatling** [comment]
Ran through all testing steps and confirmed works as expected. 👍

**Comment by p-jackson** [comment]
@andrewserong I think merging as-is without the passwordless user step is good

**Pull request merged**